src/intersection/first-solution/intersection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Creates an array of values that are in both the first and the second arrays.
+ *
+ * @param {Array} [firstArr=[]] - First array to be compared for repeated values.
+ * @param {Array} [secondArr=[]] - Second array to be compared for repeated values.
+ * @returns {Array} Returns the new array with only intersected values.
+ */
+
+export const arrayIntersection = (firstArr = [], secondArr = []) => {
+  let newArrayIntersected = [];
+  // Test if the arguments are arrays and make sure that arguments are passed
+  if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+    throw new TypeError('The arguments must be arrays');
+  } else if (!firstArr.length || !secondArr.length) {
+    throw new Error('Must pas two arrays to intercept the values');
+  } else {
+    // test if arrays have the same values, return only the values matched
+    newArrayIntersected = [...new Set(firstArr.filter((x) => secondArr.includes(x)))];
```

Comment:
No need to use set since filter method returns a new array:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

So no need to convert array to set then again to array. Unnecessary use of cpu.

You can visually see and play with it here:
http://www.pythontutor.com/live.html#code=const%20firstArr%20%3D%20%5B1,2%5D%3B%0Aconst%20secondArr%20%3D%20%5B2,3%5D%3B%0A%0Aconst%20theFilteredArray%20%3D%20firstArr.filter%28%28x%29%20%3D%3E%20secondArr.includes%28x%29%29%3B%0A%0AfirstArr%5B1%5D%20%3D%200%3B%0AsecondArr%5B0%5D%20%3D%200%3B%0A%0AtheFilteredArray.push%285%29%3B%0A%0Aconst%20theSet%20%3D%20new%20Set%28theFilteredArray%29%3B%0A%0A%0A%0Aconst%20newArrayIntersected%20%3D%20%5B...theSet%5D%3B&cumulative=false&curInstr=13&heapPrimitives=nevernest&mode=display&origin=opt-live.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false

Reply:
Is exactly the same, the difference is that you expand the code in several lines to understand what I did which is good, if you read my implementation, is well explained that line of code